pkg/devicetemplateconverter/profile_fetcher.go

```diff
@@ -0,0 +1,57 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devicetemplateconverter
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/cluster"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"google.golang.org/grpc"
+)
+
+// ProfileFetcher handles the validation and fetching of end device profile in the Device Repository.
+type ProfileFetcher struct {
+	GetPeerConn func(ctx context.Context, role ttnpb.ClusterRole, ids cluster.EntityIdentifiers) (*grpc.ClientConn, error)
+	CallOpts    func(context.Context) ([]grpc.CallOption, error)
+}
+
+// ValidateFields dictactes if the end device has the necessary fields to fetch its profile.
+func (pf *ProfileFetcher) ValidateFields(device *ttnpb.EndDevice) bool {
```

Comment:
This method name indicates it's about validation and about validation by fields. This is not what this method does, right? This just indicates whether the profile is set?

Reply:
Effectively there is validation of end-device's fields that are necessary for fetching its profile from the DR. I assume it would be more intuitive for it to be named something like `IsFetchable`.

Reply:
`IsFetchable` gives the impression that the fetcher (the receiver) is 'fetchable'. `ShouldFetchProfile` is most likely closer to what you want.